modules/dreamview/backend/simulation_world/simulation_world_service.cc

```diff
@@ -551,10 +552,49 @@ Object &SimulationWorldService::CreateWorldObjectIfAbsent(
     SetObstacleInfo(obstacle, &world_obj);
     SetObstaclePolygon(obstacle, &world_obj);
     SetObstacleType(obstacle, &world_obj);
+    SetObstacleSensor(obstacle, &world_obj);
   }
   return obj_map_[id];
 }
 
+Object &SimulationWorldService::CreateWorldObjectBySensor(
+  const SensorMeasurement &sensor) {
+  // Create a new world object and put it into object map to store sensor
+  // measurement info , sensor measurement theta——>object heading
+  Object sensor_object;
+  sensor_object.set_id(std::to_string(sensor.id()));
+  Point3D pos = sensor.position();
+  sensor_object.set_allocated_position(&pos);
+  sensor_object.set_heading(sensor.theta());
+  sensor_object.set_length(sensor.length());
+  sensor_object.set_width(sensor.width());
+  sensor_object.set_height(sensor.height());
+  sensor_object.set_sub_type(sensor.sub_type());
+  switch (sensor.type()) {
```

Comment:
for the switch part, it would be nice if we reuse SetObstacleType() from line 135.